### ISSUE TITLE
path,single-quoted-string style combination regressed

### ISSUE THREAD

**Issue by danielshahaf** [zsh-users/zsh-syntax-highlighting#707]
As #475 pointed out, when `./foo` existed, `: 'f'oo` highlighted the `'f'` in `underline,fg=yellow` and the `oo` in `underline`.  That regressed — bisected to 6835121eafa49899b396f5f85207fb1fe229bc1c.  It'd be nice to have it back ☺

**Comment by danielshahaf** [comment]
In 6835121: `typeset -a region_highlight=( '0 1 fg=green' '2 7 none' '2 7 underline' '2 5 fg=yellow' '2 5 fg=yellow' )`

In the preceding revision: `region_highlight=( '0 1 fg=green' '2 7 none' '2 5 fg=yellow' '2 7 underline' )`

**Comment by danielshahaf** [comment]
The fix is "simple":

```diff
diff --git a/highlighters/main/main-highlighter.zsh b/highlighters/main/main-highlighter.zsh
index c44b1c3..1c6768e 100644
--- a/highlighters/main/main-highlighter.zsh
+++ b/highlighters/main/main-highlighter.zsh
@@ -1314,7 +1314,7 @@ _zsh_highlight_main_highlighter_highlight_argument()
     fi
   fi
 
-  highlights=($(( start_pos + $1 - 1 )) $end_pos $base_style $highlights)
+  highlights+=($(( start_pos + $1 - 1 )) $end_pos $base_style)
   _zsh_highlight_main_add_many_region_highlights $highlights
 }
 
```

It fixes the issue, but breaks a number of tests because of ordering of expectations.

Also, I suspect it was written this way for a reason… Matthew?

**Comment by phy1729** [comment]
The reason is later styles override earlier styles, so in the case of
```
ZSH_HIGHLIGHT_STYLES[path]=fg=red
ZSH_HIGHLIGHT_STYLES[single-quoted-argument]=fg=cyan
```

`'f'` should be cyan and `oo` should be red. With the above patch, the whole string is red.

I think this is a bug in ZLE.

**Comment by danielshahaf** [comment]
Agreed. Thanks. So the fix'll be to the `$region_highlight` combiner in zle.

Labelled "good first issue", but this requires C work.
